src/allNext.ts

```diff
@@ -0,0 +1,48 @@
+import { IPaginated } from '../models/api'
+import type { Discojs } from './discojs'
+
+export class AllNext {
```

Comment:
I felt this implementation was a bit hard to follow, as you first have to call a method, then call `client.all`.
Instead of doing that, I introduced dedicated `all` methods that returns async iterators:
* `getAllMasterVersions`
* `getAllArtistReleases`
* `getAllLabelReleases`
* `getAllRecentExports`
* `getAllInventoryForUser`
* `getAllInventory`
* `listAllOrders`
* `listAllItemsByReleaseForUser`
* `listAllItemsByRelease`
* `listAllItemsInFolderForUser`
* `listAllItemsInFolder`
* `getAllSubmissionsForUser`
* `getAllSubmissions`
* `getAllContributionsForUser`
* `getAllContributions`
* `getAllListsForUser`
* `getAllLists`
* `getAllWantlistForUser`
* `getAllWantlist`

The example you have in the PR description would become:
```ts
async function updateCollection() {
  for await (const { releases } of client.listAllItemsInFolder()) {
    for (const release of releases) {
      await addToCollection(release)
    }
  }
}
```

Reply:
That seems much more user-friendly, if less magical. 😉

I probably chose my method based on writing the least code and exercising some TypeScript-fu.